libvast/src/table_slice.cpp

```diff
@@ -78,12 +78,16 @@ data_view table_slice::row_view::operator[](size_t column) const {
 table_slice::table_slice(table_slice_header header)
   : header_{std::move(header)} {
   // nop
+  ++instance_count_;
 }
 
 table_slice::~table_slice() {
   // no
+  --instance_count_;
 }
 
+std::atomic<int> table_slice::instance_count_{0};
```

Comment:
Is this supposed to be part of this PR? There doesnt seem to be a connection to jemalloc.

Also, this feature should probably get an extremely simple unit test exercising `table_slice::instances()`.

Reply:
This is in here because I started this branch when working on a memory related bug, that work stalled but the branch still existed, so I reactivated it.